sdlib/d/gc/slab.d

```diff
@@ -98,6 +98,49 @@ unittest SlabRange {
 	assert(r0.length == 12);
 }
 
+// encapsulate metadata reading, and private setting.
+struct SlabMetadata {
```

Comment:
It is generally not a good sign when you need to describe what the struct is. Either it is poorly named, or the concept itself is ill defined.

Sentences start with a capital letter and end with a dot.

I'm a bit skeptical of the design here. This abstraction was added, yet the code that uses it also got larger.

The cod doesn't follow the convention found elsewhere in the codebase, such as fields being defined at the top of structs.

The name `SlabMetadata` is already used within extents. Not a conflict per se, but another indicator that the name might be generic.

Overall, I'm a bit doubtful that this is the right abstraction to tackle the problem at hand.

Reply:
> It is generally not a good sign when you need to describe what the struct is. Either it is poorly named, or the concept itself is ill defined.

I can remove the comment, I think the name is self explanatory. I'm also fine changing the name, it doesn't matter too much to me.

> I'm a bit skeptical of the design here. This abstraction was added, yet the code that uses it also got larger.

Part of the reason the code is larger because SDC doesn't allow member calls on rvalues. You can see the code I *wanted* to write commented out. So subtract 28 lines due to that bug.

Also the sdc format requirements for a property function require 4 lines for a 1-line function. I can save 2 lines by doing the alias thing that was in the original, but I found that technique to be unworthy of the obfuscation it adds. If lines of code added is a more important metric, I can do it.

Also, I don't know how you add an abstraction to be used elsewhere without adding lines of code. The savings is in the code you didn't have to write where you use it (in this case in the collection routines).

> The cod doesn't follow the convention found elsewhere in the codebase, such as fields being defined at the top of structs.

OK.

> Overall, I'm a bit doubtful that this is the right abstraction to tackle the problem at hand.

What abstraction do you think is best? I am trying to avoid the copy/paste that I had in the original PR, and encapsulate all the functionality inside slab.d without having to construct a SlabAllocInfo (If you recall, this was a performance problem).

For reference, here is the commit from the original PR that basically details all the changes I had made for slab finalizers: https://github.com/snazzy-d/sdc/pull/368/commits/75ba6425fb8785f6eefe7855615510c61b082ef6